tensorflow/contrib/crf/python/kernel_tests/crf_test.py

```diff
@@ -199,6 +200,87 @@ def testViterbiDecode(self):
       self.assertEqual(actual_max_sequence,
                        expected_max_sequence[:sequence_lengths])
 
+  def testCrfDecode(self):
+
+    def np_crf_decode(potentials, transition_params, seq_lens):
+      '''This is a function for numpy arrays.
+
+      Args:
+        potentials: [B, T, num_tags]
+        transition_params: [num_tags, num_tags]
+        seq_lens: [B]
+        tag_indices: [B, T]
+
+      Returns:
+        pred_tags: a list
+        scores: a list
+      '''
+      pred_tags, scores = [], []
+      for p, seq_len in zip(potentials, seq_lens):
+        p = p[:seq_len]
+        pred_tag, score = crf.viterbi_decode(p, transition_params)
+        pred_tags.append(pred_tag)
+        scores.append(score)
+      return pred_tags, scores
+
+
+    def calc_accuracy(pred_tags, gold_tags, seq_lens):
+      correct, total = 0, 0
+      for pred, gold, seq_len in zip(pred_tags, gold_tags, seq_lens):
+        pred = pred[:seq_len]
+        gold = gold[:seq_len]
+        correct += np.sum(np.equal(pred, gold))
+        total += seq_len
+      accuracy = correct / total
+      return accuracy
+
+    batch_size = 32
+    max_seq_len = 100
+    num_tags = 4
+
+    # np.random.seed(1)
```

Comment:
I would suggest leaving that in. It's easier to debug if the tests are deterministic.